### ISSUE TITLE
Connect to a kerberized kafka cluster

### ISSUE THREAD

**Issue by wangandi** [MaterializeInc/materialize#2303]
~Possibly related to #1283~
Possibly related to #1989

**Comment by rjnn** [comment]
Here's the MVP configurations that we need to support:

“security.protocol”: “sasl_plaintext”
“sasl.mechanisms”: “GSSAPI” (this is the default too)
“sasl.kerberos.service.name”: “kafka” (this is the default too)

From the prospect, the easiest would be if materialize could just expose kafka consumer configs (maybe a filtered set) as part of the `CREATE SOURCE FROM KAFKA BROKER`, and then the user can specify the parameters that will get passed through.

If we do add support for keytab files, we should support the following fields:
* “sasl.kerberos.principal”: the principal of the keytab, we provide it
* “sasl.kerberos.keytab”: the path to the keytab file, we provide it
* “sasl.kerberos.kinit.cmd”
* “sasl.kerberos.min.time.before.relogin”

 
Do note that there is a known annoyance with librdkafka where it periodically refreshes the TGT token with kinit, and so the prospect does wish to proactively set `sasl.kerberos.kinit.cmd` and/or `sasl.kerberos.min.time.before.relogin` so they would appreciate support for that.

**Comment by sploiselle** [comment]
Running into some error-handling issues:

```
materialize=> CREATE SOURCE quotes
materialize->   FROM KAFKA BROKER 'kafka.sean-VirtualBox:9093' TOPIC 'test'
materialize->   WITH (
materialize(>     security_protocol = 'sasl_plaintext',
materialize(>     sasl_kerberos_keytab = '/etc/materialize.keytab',
materialize(>     sasl_kerberos_principal = 'materialize/kafka.sean-VirtualBox:9093@sean-VirtualBox'
materialize(>   )
materialize->   FORMAT BYTES;
```
```
materialize=> select * from quotes;
ERROR:  Unable to automatically determine a timestamp for your query; this can happen if your query depends on non-materialized sources
```

but the actual error I'm encountering is that the auth is failing:
```
kinit: Client 'materialize/kafka.sean-VirtualBox:9093@sean-VirtualBox' not found in Kerberos database while getting initial credentials
```

i.e. the error has nothing to do with timestamps, it's an auth failure

**Comment by sploiselle** [comment]
Closed  as of #2870
